modules/kameleon/src/kameleonhelper.cpp

```diff
@@ -69,97 +69,97 @@ std::unique_ptr<ccmc::Kameleon> createKameleonObject(const std::string& cdfFileP
-                    case 17: // YYYY-MM-DDTHH:MM:      =>       YYYY-MM-DDTHH:MM:SS
-                        seqStartStr += "00";
-                        [[fallthrough]];
-                    // case 19 : // YYYY-MM-DDTHH:MM:SS    =>    YYYY-MM-DDTHH:MM:SS.000
-                    //     seqStartStr += ".000";
-                    // case 23 : // YYYY-MM-DDTHH:MM:SS.   =>    YYYY-MM-DDTHH:MM:SS.000Z
-                    //     seqStartStr += "Z";
-                    default:
-                        break;
-                }
+    std::string seqStartStr;
+    if (kameleon->doesAttributeExist("start_time")){
+        seqStartStr =
+                kameleon->getGlobalAttribute("start_time").getAttributeString();
+    }
+    else if (kameleon->doesAttributeExist("tim_rundate_cal")) {
+        seqStartStr =
+                kameleon->getGlobalAttribute("tim_rundate_cal").getAttributeString();
+        const size_t N_CHARS = seqStartStr.length();
+        if (N_CHARS < 19) {
+            // Fall through to add the required characters
+            switch (N_CHARS) {
+                case 10: // YYYY-MM-DD             =>       YYYY-MM-DDTHH
+                    seqStartStr += "T00";
+                    [[fallthrough]];
+                case 13: // YYYY-MM-DDTHH          =>       YYYY-MM-DDTHH:
+                    seqStartStr += ":";
+                    [[fallthrough]];
+                case 14: // YYYY-MM-DDTHH:         =>       YYYY-MM-DDTHH:MM
+                    seqStartStr += "00";
+                    [[fallthrough]];
+                case 16: // YYYY-MM-DDTHH:MM       =>       YYYY-MM-DDTHH:MM:
+                    seqStartStr += ":";
+                    [[fallthrough]];
+                case 17: // YYYY-MM-DDTHH:MM:      =>       YYYY-MM-DDTHH:MM:SS
+                    seqStartStr += "00";
+                    [[fallthrough]];
+                // case 19 : // YYYY-MM-DDTHH:MM:SS    =>    YYYY-MM-DDTHH:MM:SS.000
+                //     seqStartStr += ".000";
+                // case 23 : // YYYY-MM-DDTHH:MM:SS.   =>    YYYY-MM-DDTHH:MM:SS.000Z
+                //     seqStartStr += "Z";
+                default:
+                    break;
             }
         }
-        else if (kameleon->doesAttributeExist("tim_obsdate_cal")) {
-            seqStartStr =
-                    kameleon->getGlobalAttribute("tim_obsdate_cal").getAttributeString();
-        }
-        else if (kameleon->doesAttributeExist("tim_crstart_cal")) {
-            seqStartStr =
-                    kameleon->getGlobalAttribute("tim_crstart_cal").getAttributeString();
-        }
-        else {
-            LWARNING(
-                "No starting time attribute could be found in the .cdf file. Starting "
-                "time is set to 01.JAN.2000 12:00."
-            );
-        }
+    }
+    else if (kameleon->doesAttributeExist("tim_obsdate_cal")) {
+        seqStartStr =
+                kameleon->getGlobalAttribute("tim_obsdate_cal").getAttributeString();
```

Comment:
again, super nitpicky, but this seem to be intented twice when once should be enough (same for next else if)